From 38d02d71ae2ce9dd4e7165c6b446589627175967 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sat, 7 Jun 2014 15:59:23 +0100
Subject: [PATCH 5/6] build: Try both lib and lib64, also skip "not found" deps

Under Evolve OS, for some unknown reason the "not found" string with
random dependencies will cause gobject-introspection to become a dependency
of the package.

This change ensures there is a linked library to search for now, as well
as ensuring that "/lib/" and "/lib64/" links both work for the same
packages. Eventually this compatibility will disappear when all packages
within Evolve OS have been forced into /usr/lib/
---
 pisi/operations/build.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pisi/operations/build.py b/pisi/operations/build.py
index 0a7ab24..1bc2c12 100644
--- a/pisi/operations/build.py
+++ b/pisi/operations/build.py
@@ -935,6 +935,8 @@ class Builder:
             for line in out.split("\n"):
                 line = line.strip().rstrip()
                 if "=>" in line:
+                    if "not found" in line:
+                        continue
                     dep = line.split("=>")[1]
                     dep = dep.strip().rstrip().split()[0]
 
@@ -944,7 +946,10 @@ class Builder:
                     if os.path.exists(tpath) or os.path.exists(tpath2):
                         continue
 
-                    dep2 = dep.replace("/lib/", "/lib64/")
+                    if "lib64" in dep:
+                        dep2 = dep.replace("/lib64/", "/lib/")
+                    else:
+                        dep2 = dep.replace("/lib/", "/lib64/")
                     result = pisi.api.search_file(dep)
                     if not result:
                         result = pisi.api.search_file(dep2)
-- 
1.9.1

